components/button/button-add.js

```diff
@@ -59,30 +79,131 @@ class ButtonAdd extends PropertyRequiredMixin(FocusMixin(LocalizeCoreElement(Lit
 			button:${unsafeCSS(getFocusPseudoClass())} .line {
 				border-top-color: var(--d2l-color-celestine);
 			}
+		`;
+	}
 
-			.content {
+	constructor() {
+		super();
+
+		this.hintNearbyDistance = DEFAULT_HINT_DISTANCE;
+		this.hintTiming = HINT_TIMING.ALWAYS;
+		this.textVisible = false;
+
+		this._buttonId = getUniqueId();
+		this._hasFocusOrHover = false;
+		this._visibleDistance = 0;
+
+		this.addEventListener('focus', this._onFocusHover);
+		this.addEventListener('mouseover', this._onFocusHover);
+		this.addEventListener('blur', this._onBlur);
+		this.addEventListener('mouseout', this._onBlur);
+	}
+
+	static get focusElementSelector() {
+		return 'button';
+	}
+
+	render() {
+		const buttonSpacing = this._getButtonSpacing();
+		const text = this.text || this.localize('components.button-add.addItem');
+		const content = this.textVisible ? this._renderWithTextVisible(text) : this._renderWithTextHidden(text);
+		const id = !this.textVisible ? this._buttonId : undefined;
+
+		return html`
+			<button
+				class="d2l-label-text d2l-visible-on-ancestor-target"
+				id="${ifDefined(id)}"
+				style="${styleMap(buttonSpacing)}"
+				type="button">
+				<div class="line"></div>
+				${content}
+			</button>
+		`;
+	}
+
+	_getButtonSpacing() {
+		if (this.textVisible || this.hintTiming !== HINT_TIMING.NEARBY) return {};
+
+		this._visibleDistance = this.hintNearbyDistance || DEFAULT_HINT_DISTANCE;
+		return {
+			marginBottom: `-${this._visibleDistance}px`,
```

Comment:
I think we briefly talked about this when we met with Jeff, but I'll raise it again since I think it might bite us. Overlapping with other elements can be problematic. In this case, the link is entirely non-operable with a mouse.

![image](https://github.com/BrightspaceUI/core/assets/9042472/bd9df29f-9d4a-4928-97b1-521d39fa27fa)

Reply:
Yeah agreed. I ran into this when trying to add it into list - it basically seems (which makes sense) like you can get either the interactive element (e.g., list item) or the link but not both. We likely want to rethink having `hintNearbyDistance` at all and just have the hover distance be whatever `button-add`'s padding is. I'll raise this with Sravan.